test-app/backend/src/shift/shift-api.ts

```diff
@@ -0,0 +1,45 @@
+import app from '../server';
```

Comment:
One thing I notice here is that while you are doing good error checking, there are still some assumptions. For the start/end dates for instance, you check if `start >= end`. But what if `start` or `end` is actually not even defined. We also need checks for that sort of thing, like with what you did for title. `if (!start || !end || start >= end) <some_error>`. There are other places that need this checked too.